include/board.c

```diff
@@ -1,33 +1,91 @@
-#include<stdio.h>
+#include <stdio.h>
 #include "board.h"
 
-void init_board(struct Board *board) {
-    for (int i = 0; i < SIZE; i++) {
-        for (int j = 0; j < SIZE; j++) {
+void init_board(struct Board *board)
+{
+    for (int i = 0; i < SIZE; i++)
+    {
+        for (int j = 0; j < SIZE; j++)
+        {
             board->board[i][j] = EMPTY;
         }
     }
 }
 
-
-void print_board(struct Board *board) {
+void print_board(struct Board *board)
+{
     printf("\n");
     printf("    ");
-    for (int col = 0; col < SIZE; col++) {
+    for (int col = 0; col < SIZE; col++)
+    {
         printf("%d ", col);
     }
     printf("\n");
     printf("    ");
-    for (int col = 0; col < SIZE; col++) {
+    for (int col = 0; col < SIZE; col++)
+    {
         printf("--");
     }
     printf("\n");
-    for (int row = 0; row < SIZE; row++) {
+    for (int row = 0; row < SIZE; row++)
+    {
         printf("%d | ", row);
-        for (int col = 0; col < SIZE; col++) {
+        for (int col = 0; col < SIZE; col++)
+        {
             printf("%c ", board->board[row][col]);
         }
         printf("\n");
     }
     printf("\n");
-}
\ No newline at end of file
+}
+
+int check_draw(struct Board *board) // 0: no draw, 1: draw
+{
+    for (int row = 0; row < 3; row++)
+    {
+        for (int col = 0; col < 3; col++)
+        {
+            if (board->board[row][col] != 'X' && board->board[row][col] != 'O')
+            {
+                return 0;
+            }
+        }
+    }
+    return 1;
+}
+
+int check_win(struct Board *board) // 0: no winner, 1: player wins
+{
+    // Check rows
```

Comment:
Modify it to return the player who wins instead of just checking if someone won.

Reply:
> Modify it to return the player who wins instead of just checking if someone won.

`int check_board(struct Board *board) // 0: no winner, 1: X wins, 2: O wins, 3: draw`

this exception is handled in the check_board()

Reply:
That's what I'm saying; we don't need the check_bord function. We can just return a char in the check_win function (maybe rename to something more suitable)